### PR TITLE
Ww 255

### PR DESCRIPTION
**Pull request by cormorando**
https://wikia-inc.atlassian.net/browse/WW-255

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/53f809ba5c8dbbf63e6a136991da41fbeca1630f/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L44](https://github.com/Wikia/app/blob/53f809ba5c8dbbf63e6a136991da41fbeca1630f/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L44): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L295](https://github.com/Wikia/app/blob/53f809ba5c8dbbf63e6a136991da41fbeca1630f/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L295): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php [hunk 1]
```diff
@@ -36,6 +36,7 @@ public function index() {
 			'heroImageUrl' => $this->getHeroImageUrl(),
 			'inviteFriendsText' => $this->msg( 'communitypage-invite-friends' )->plain(),
 			'headerWelcomeMsg' => $this->msg( 'communitypage-tasks-header-welcome', $wgWikiTopic ?? $wgSitename )->parse(),
+			'subheaderWelcomeMsg' => $this->msg( 'communitypage-subheader-welcome' )->text(),
 			'adminWelcomeMsg' => $this->msg( 'communitypage-admin-welcome-message' )->text(),
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php]: let's remove `adminWelcomeMsg` from here and from .i18n.php file as well (`communitypage-admin-welcome-message`) 💃

File: extensions/wikia/CommunityPage/styles/components/_header.scss [hunk 2]
```diff
@@ -12,12 +12,14 @@ $hero-image-background: rgba(0,0,0,0.35);
...
 	@include flexbox();
-	align-items: center;
 	background: url($hero-image-pattern);
 	background-color: $hero-image-background;
 	height: $hero-image-height;
 	justify-content: center;
+	left: -30px;
+	position: relative;
 	text-align: center;
+	width: calc(100% + 60px);
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/styles/components/_header.scss]: can we move the 30px to a `$page-margin` var and use it here and above? :)

File: extensions/wikia/CommunityPage/styles/components/_header.scss [hunk 3]
```diff
@@ -45,22 +49,17 @@ $hero-image-background: rgba(0,0,0,0.35);
 		font-size: 36px;
 		font-weight: bold;
 		line-height: 48px;
+		margin:	0;
 	}
 
-	h1.community-page-heading:lang(ja) {
+	h1.community-page-subheading {
+		color: $color-white;
 		font-size: 24px;
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/styles/components/_header.scss]: what about moving it to var and use here and in l63?

File: extensions/wikia/CommunityPage/templates/pageHeader.mustache [hunk 4]
```diff
@@ -1,8 +1,6 @@
 <div class="community-page-header {{#heroImageUrl}}community-page-header-cover" style="background-image: url({{heroImageUrl}});{{/heroImageUrl}}">
 	<div class="community-page-header-content">
 		<h1 class="community-page-heading">{{{headerWelcomeMsg}}}</h1>
+		<h1 class="community-page-subheading">{{{subheaderWelcomeMsg}}}</h1>
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/templates/pageHeader.mustache]: pages shouldn't have more than one `h1` - please use `h2` or `h3` as it is a subheader, not a main one

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/fc0ba2b2b430473f03fe05b65b7bbd9a185b5955/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L43](https://github.com/Wikia/app/blob/fc0ba2b2b430473f03fe05b65b7bbd9a185b5955/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L43): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L294](https://github.com/Wikia/app/blob/fc0ba2b2b430473f03fe05b65b7bbd9a185b5955/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L294): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/CommunityPage/styles/components/_header.scss (hunk 3 above)
Comment (cormorando) [reply extensions/wikia/CommunityPage/styles/components/_header.scss]: I'd prefer not - they're not really correlated

**Comment by lukaszkonieczny** [comment]
LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/7af906d7cd9255157dde05873ecfa4fa52dc8571/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L43](https://github.com/Wikia/app/blob/7af906d7cd9255157dde05873ecfa4fa52dc8571/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L43): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L294](https://github.com/Wikia/app/blob/7af906d7cd9255157dde05873ecfa4fa52dc8571/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L294): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/85a104e588f80f12b27193adfac2b8ed0a3bd7ac/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L43](https://github.com/Wikia/app/blob/85a104e588f80f12b27193adfac2b8ed0a3bd7ac/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L43): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L294](https://github.com/Wikia/app/blob/85a104e588f80f12b27193adfac2b8ed0a3bd7ac/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L294): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php [hunk 5]
```diff
@@ -36,7 +36,7 @@ public function index() {
 			'heroImageUrl' => $this->getHeroImageUrl(),
 			'inviteFriendsText' => $this->msg( 'communitypage-invite-friends' )->plain(),
 			'headerWelcomeMsg' => $this->msg( 'communitypage-tasks-header-welcome', $wgWikiTopic ?? $wgSitename )->parse(),
-			'adminWelcomeMsg' => $this->msg( 'communitypage-admin-welcome-message' )->text(),
+			'subheaderWelcomeMsg' => $this->msg( 'communitypage-subheader-welcome' )->text(),
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:39]: did you synced with Lukasz about i18n strings naming? Just not to generate additional work

File: extensions/wikia/CommunityPage/styles/components/_header.scss [hunk 6]
```diff
@@ -15,8 +15,10 @@ $hero-image-background: rgba(0,0,0,0.35);
 	align-items: center;
 	background: url($hero-image-pattern);
 	background-color: $hero-image-background;
+	display: flex;
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/styles/components/_header.scss]: please use `@include flexbox();`

File: extensions/wikia/CommunityPage/styles/components/_header.scss [hunk 7]
```diff
@@ -15,8 +15,10 @@ $hero-image-background: rgba(0,0,0,0.35);
...
 	height: $hero-image-height;
 	justify-content: center;
+	position: relative;
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/styles/components/_header.scss:21]: please use `@include justify-content(center);`

File: extensions/wikia/CommunityPage/styles/components/_header.scss [hunk 8]
```diff
@@ -45,22 +48,47 @@ $hero-image-background: rgba(0,0,0,0.35);
 		font-size: 36px;
 		font-weight: bold;
 		line-height: 48px;
+		margin:	0;
+	}
+
+	h3.community-page-subheading {
+		color: $color-white;
+		font-size: 24px;
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/styles/components/_header.scss]: please include: `@import 'extensions/wikia/DesignSystem/bower_components/design-system/base/typography';` and use font-size & `line-height: $lihe-height;` from there

File: extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php (hunk 5 above)
Comment (rybmat) [reply extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:39]: I wasn't changing that, but good catch, looks like it is not included here: https://github.com/Wikia/app/pull/11322/files

Comment (rybmat) [reply extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php:39]: wait, it is: https://github.com/Wikia/app/blob/e86c5387e4b655ce0ee68d3e7fe5c25ffd3279e1/extensions/wikia/CommunityPage/CommunityPage.i18n.php#L30

File: extensions/wikia/CommunityPage/templates/pageHeader.mustache [hunk 9]
```diff
@@ -1,8 +1,6 @@
 <div class="community-page-header {{#heroImageUrl}}community-page-header-cover" style="background-image: url({{heroImageUrl}});{{/heroImageUrl}}">
 	<div class="community-page-header-content">
 		<h1 class="community-page-heading">{{{headerWelcomeMsg}}}</h1>
```
Comment (sebastianmarzjan) [inline extensions/wikia/CommunityPage/templates/pageHeader.mustache]: Not to be picky, but class and message have inconsistent naming

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/6018869ee41f325881185ce968e382074a23a2b9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L43](https://github.com/Wikia/app/blob/6018869ee41f325881185ce968e382074a23a2b9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L43): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L294](https://github.com/Wikia/app/blob/6018869ee41f325881185ce968e382074a23a2b9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L294): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by sebastianmarzjan** [comment]
LGTM

**Comment by cormorando** [comment]
LG

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/3f0724073987ac39045f65bd6ec352fe11c51630/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L43](https://github.com/Wikia/app/blob/3f0724073987ac39045f65bd6ec352fe11c51630/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L43): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L294](https://github.com/Wikia/app/blob/3f0724073987ac39045f65bd6ec352fe11c51630/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L294): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/6018869ee41f325881185ce968e382074a23a2b9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
2. MINOR [CommunityPageSpecialController.class.php#L43](https://github.com/Wikia/app/blob/6018869ee41f325881185ce968e382074a23a2b9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L43): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
3. MINOR [CommunityPageSpecialController.class.php#L294](https://github.com/Wikia/app/blob/6018869ee41f325881185ce968e382074a23a2b9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L294): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Pull request merged**